### PR TITLE
Refactored tactics handling, and added unit tests

### PR DESCRIPTION
**Pull request by eddie-knight**
summary notes:

- The Run function has reduced complexity by delegating specific operations to helper functions
- The Tactic struct and its execution logic were extracted into a new tactic.go file, further modularizing the codebase
- New tests added for Run and runTactic, including edge cases like missing tactics in the configuration
- The introduction of a testArmory mock makes testing easier by simulating the behavior of the Armory interface
- Reduced complexity of condition for multiple and single tactics
- The tests now cover cases like multiple tactics, failed tactics, and missing configurations

### CODE REVIEW

File: raidengine/raidengine.go [hunk 1]
```diff
@@ -21,136 +16,80 @@ type MovementResult struct {
...
-		name, strikeResult := strike()
-		if strikeResult.Message == "" {
-			strikeResult.Message = "Strike did not return a result, and may still be under development."
-		}
-		if strikeResult.Passed {
-			successes += 1
-			logger.Info(strikeResult.Message)
+	newErr := tactic.Execute()
+	if newErr != nil {
+		if err != nil {
```
Comment (grudra7714) [inline raidengine/raidengine.go:88]: dont think we need this if/else block, just 
```
err = fmt.Errorf("%s\n%s", err.Error(), newErr.Error())
```  
or
```
err = newErr
```
should do

File: raidengine/tactic.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			logger.Info(strikeResult.Message)
+		} else {
+			t.failures += 1
+			logger.Error(strikeResult.Message)
+		}
+		t.AddStrikeResult(name, strikeResult)
+	}
+	t.EndTime = time.Now().String()
+	t.WriteStrikeResultsJSON()
+	t.WriteStrikeResultsYAML()
```
Comment (grudra7714) [inline raidengine/tactic.go:82]: Future work: Can make a utility for this -- it can handle both write options with an optional configurational value

**Review by grudra7714 — CHANGES_REQUESTED** [review]

File: raidengine/raidengine.go (hunk 1 above)
Comment (eddie-knight) [reply raidengine/raidengine.go:88]: I'll push that up now

File: raidengine/tactic.go (hunk 2 above)
Comment (eddie-knight) [reply raidengine/tactic.go:82]: good idea

**Review by grudra7714 — APPROVED** [review]

**Review by grudra7714 — APPROVED** [review]
